src/main/java/no/finn/unleash/repository/FeatureToggleRepository.java

```diff
@@ -40,7 +39,14 @@ public FeatureToggleRepository(
         this.toggleFetcher = toggleFetcher;
         this.eventDispatcher = new EventDispatcher(unleashConfig);
 
-        toggleCollection = toggleBackupHandler.read();
+        this.toggleCollection = toggleBackupHandler.read();
```

Comment:
So we are 100% sure toggleBackupHandler cannot return null?

Reply:
It can absolutely return null, I've annotated the backupHandler method to signify this. Just had quite a few tests fail because I removed the check on L43 as to whether or not it was null.

Reply:
Though that does seem to be due to the fact that we mock the entire class, Looking at the code for BackupHandler it seems to always return a ToggleCollection, either with feature toggles or with an empty feature toggle list.